File: openapi-generator/scalaz/src/main/scala/org/openapitools/client/api/AssetReeferResponseReeferStatsFuelPercentage.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import AssetReeferResponseReeferStatsFuelPercentage._

case class AssetReeferResponseReeferStatsFuelPercentage (
  /* Fuel percentage of the reefer. */
  fuelPercentage: Option[Long],
/* Timestamp in Unix milliseconds since epoch. */
  changedAtMs: Option[Long])

object AssetReeferResponseReeferStatsFuelPercentage {
  import DateTimeCodecs._

  implicit val AssetReeferResponseReeferStatsFuelPercentageCodecJson: CodecJson[AssetReeferResponseReeferStatsFuelPercentage] = CodecJson.derive[AssetReeferResponseReeferStatsFuelPercentage]
  implicit val AssetReeferResponseReeferStatsFuelPercentageDecoder: EntityDecoder[AssetReeferResponseReeferStatsFuelPercentage] = jsonOf[AssetReeferResponseReeferStatsFuelPercentage]
  implicit val AssetReeferResponseReeferStatsFuelPercentageEncoder: EntityEncoder[AssetReeferResponseReeferStatsFuelPercentage] = jsonEncoderOf[AssetReeferResponseReeferStatsFuelPercentage]
}

File: openapi-generator/dart-jaguar/lib/model/asset_reefer_response_reefer_stats_fuel_percentage.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

part 'asset_reefer_response_reefer_stats_fuel_percentage.jser.dart';

class AssetReeferResponseReeferStatsFuelPercentage {
   /* Fuel percentage of the reefer. */
  @Alias('fuelPercentage')
  final int fuelPercentage;
   /* Timestamp in Unix milliseconds since epoch. */
  @Alias('changedAtMs')
  final int changedAtMs;
  

  AssetReeferResponseReeferStatsFuelPercentage(
      

{
     this.fuelPercentage = null,  
     this.changedAtMs = null 
    
    }
  );

  @override
  String toString() {
    return 'AssetReeferResponseReeferStatsFuelPercentage[fuelPercentage=$fuelPercentage, changedAtMs=$changedAtMs, ]';
  }
}

@GenSerializer(nullableFields: true)
class AssetReeferResponseReeferStatsFuelPercentageSerializer extends Serializer<AssetReeferResponseReeferStatsFuelPercentage> with _$AssetReeferResponseReeferStatsFuelPercentageSerializer {

}
